### DISCUSSION TITLE
automate slack notification

### DISCUSSION THREAD

**Discussion by samg666** [Budibase/budibase#3517] [Help] [answered]
Hello,

I have a ticketing system and would like to trigger slack notification once anyone made a reply(click the reply button). Then I have created the automate on Budibase and I am able to receive message on Slack when I ran "Finish and test automation". 

But when trigger by a button actions, it always got error `{"message":"Database does not exist.","status":404}` on the console. I am not sure what is wrong. Appreciate if anyone can help.

This is my automate setup, I am able to send message to Slack by "Finish and test automation".
![iShot2021-11-24 16 06 39](https://user-images.githubusercontent.com/29649651/143200101-013f8aa8-789d-4d7c-b395-8600d1a78145.png)

This is my button action setup, always got error when triggered Slack notification.
![iShot2021-11-24 16 07 19](https://user-images.githubusercontent.com/29649651/143200167-6c4997d3-0a47-4dda-9d0f-b345bca9deb6.png)

**Comment by mjashanks** [comment, marked as answer]
Hi Sam. Have you published your app?

Automations only work in
1. Test mode (as you tried above)
2. Published apps

We don't want automations running in dev, as it could have a lot of unintended consequences!

**Reply by samg666** [reply]
@mjashanks  thanks Michael, it works on public apps. But I always got message `{"Message":"test"} ` on slack, how can I just show the message `test` instead of JSON on slack?

**Reply by mjashanks** [reply]
I just tried this - I set my message text to "Hello", and I saw "Hello" in Slack

**Reply by samg666** [reply]
@mjashanks  Can you share your field setup on automate? I think maybe I used a wrong field setup.

**Reply by samg666** [reply]
I have solved this issue, the Message should reference to `{{ trigger.fields.Fieldname }}` then it will just pass the text from the frontend.
![image](https://user-images.githubusercontent.com/29649651/143373304-8039eeef-6b61-469a-8c77-8da10a1f5ac0.png)
